Assets/Scripts/MovementModelController.cs

```diff
@@ -0,0 +1,64 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+public class MovementModelController
+{
+
+    private CandyPool _candyPool;
+    private GridManagerViewer _gridManager;
+    private GameSettings _gameSettings;
+    private MatchHandlerViewer _matchHandler;
+    public CandyViewer SelectedCandy { get; set; }
+    private MovementViewer _movementViewer;
+    public bool IsMoving { get; set; } = false;
+    private bool isActive1 = false;
+ 
+
+    public MovementModelController(GridManagerViewer gridManager, GameSettings gameSettings, CandyPool candyPool, MatchHandlerViewer matchHandler, MovementViewer movementViewer)
+    {
+        _gridManager = gridManager;
+        _gameSettings = gameSettings;
+        _candyPool = candyPool;
+        _matchHandler = matchHandler;
+        _movementViewer = movementViewer;
+    }
+
+    public void OnMovePerformed(InputAction.CallbackContext context/*, CandyViewer selectedCandy*/)
```

Comment:
Since the purpose of a Model or Controller script is for it to be testable, you should always think about it when keeping methods and logic here. For example, do you think it's possible to test this method? What problems would you face if you tried to write a test for this? As an exercise, try to write one for this method and see what errors you face